src/hyperlink/_url.py

```diff
@@ -19,36 +19,50 @@
 import sys
 import string
 import socket
-from typing import Callable, Text, Type
+from socket import AF_INET, AF_INET6
+try:
+    from socket import AddressFamily
+except ImportError:
+    AddressFamily = int  # type: ignore[assignment,misc] Python 2
+from typing import (
+    Any, Callable, Dict, Iterable, Iterator, List, Mapping, Optional,
+    Sequence, Text, Tuple, Type, TypeVar, Union, cast,
+)
 from unicodedata import normalize
 from ._socket import inet_pton
 try:
-    from collections.abc import Mapping
+    from collections.abc import Mapping as MappingABC
 except ImportError:  # Python 2
-    from collections import Mapping
+    from collections import Mapping as MappingABC
 
 from idna import encode as idna_encode, decode as idna_decode
 
 
 PY2 = (sys.version_info[0] == 2)
-unicode = type(u'')
```

Comment:
`typing.Text` is `unicode` on Py2 and `str` on Py3.  Since we're importing that now, we don't need this.